File: src/protocol/prompts/chooseDamageAssignmentOrder.ts
import type { GameCard } from "@/types/manabrew";

export type Type = "chooseDamageAssignmentOrder";
export type Input = {
  type: Type;
  attackerId: string;
  blockerIds: string[];
  blockerCards: GameCard[];
};
export type Output = {
  type: "damageAssignmentOrderDecision";
  orderedBlockerIds: string[];
};

File: manabrew-rs/crates/manabrew-protocol/src/prompts/choose_damage_assignment_order.rs
use serde::{Deserialize, Serialize};
use ts_rs::TS;

use crate::values::CardDto;

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(rename_all = "camelCase")]
#[ts(export, export_to = "prompts/chooseDamageAssignmentOrder.ts")]
pub struct ChooseDamageAssignmentOrderInput {
    pub attacker_id: String,
    pub blocker_ids: Vec<String>,
    pub blocker_cards: Vec<CardDto>,
}

#[derive(Debug, Clone, Serialize, Deserialize, TS)]
#[serde(
    tag = "type",
    rename_all = "camelCase",
    rename_all_fields = "camelCase"
)]
#[ts(export, export_to = "prompts/chooseDamageAssignmentOrder.ts")]
pub enum ChooseDamageAssignmentOrderOutput {
    DamageAssignmentOrderDecision { ordered_blocker_ids: Vec<String> },
}
